javalin/src/test/java/io/javalin/TestBeforeAfterMatched.kt

```diff
@@ -0,0 +1,331 @@
+package io.javalin
+
+import io.javalin.http.ContentType
+import io.javalin.http.HttpStatus
+import io.javalin.http.servlet.DefaultTasks.AFTER
+import io.javalin.http.servlet.DefaultTasks.AFTER_MATCHED
+import io.javalin.http.servlet.DefaultTasks.BEFORE
+import io.javalin.http.servlet.DefaultTasks.ERROR
+import io.javalin.http.servlet.DefaultTasks.HTTP
+import io.javalin.http.staticfiles.Location
+import io.javalin.testing.TestUtil
+import org.assertj.core.api.Assertions.assertThat
+import org.eclipse.jetty.server.handler.ContextHandler
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+class TestBeforeAfterMatched {
+
+    @Test
+    fun `beforeMatched and afterMatched work`() = TestUtil.test { app, http ->
+        app.before { ctx ->
+            ctx.result("foo")
+        }
+        app.beforeMatched { ctx ->
+            ctx.result("before-matched")
+        }
+        app.get("/hello") { ctx ->
+            ctx.result(ctx.result() + "-hello")
+        }
+        app.afterMatched { ctx ->
+            ctx.result(ctx.result() + "-after-matched")
+        }
+        app.after { ctx ->
+            ctx.result(ctx.result() + "!")
+        }
```

Comment:
Just a note, but I think it'd be nice to reduce the spacing in this test class by using one-liners with `it` here:

```
app.before { it.result("foo") }
app.beforeMatched { it.result("before-matched") }
```

Reply:
Done.